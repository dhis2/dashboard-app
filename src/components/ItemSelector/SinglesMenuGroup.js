import React, { Fragment } from 'react';
import { connect } from 'react-redux';

import HeaderMenuItem from './HeaderMenuItem';
import ContentMenuItem from './ContentMenuItem';
import { acAddDashboardItem } from '../../actions/editDashboard';

export const SinglesMenuGroup = ({ acAddDashboardItem, category }) => {
    const addToDashboard = ({ type, content }) => () => {
        acAddDashboardItem({ type, content });
    };

    return (
        <Fragment>
            <HeaderMenuItem title={category.header} />
            {category.items.map(item => (
                <ContentMenuItem
                    key={item.type}
                    type={item.type}
                    name={item.name}
                    onInsert={addToDashboard(item)}
                />
            ))}
        </Fragment>
    );
};

export default connect(
    null,
    { acAddDashboardItem }
)(SinglesMenuGroup);
