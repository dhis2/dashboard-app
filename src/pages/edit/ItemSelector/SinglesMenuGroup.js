import PropTypes from 'prop-types'
import React from 'react'
import { connect } from 'react-redux'
import { tSetDashboardItems } from '../../../actions/editDashboard'
import ContentMenuItem from './ContentMenuItem'
import HeaderMenuItem from './HeaderMenuItem'

const SinglesMenuGroup = ({ onAddItem, category }) => {
    const addToDashboard =
        ({ type, content }) =>
        () => {
            onAddItem({ type, content })
        }

    return (
        <>
            <HeaderMenuItem title={category.header} />
            {category.items.map(item => (
                <ContentMenuItem
                    key={item.type}
                    type={item.type}
                    name={item.name}
                    onInsert={addToDashboard(item)}
                />
            ))}
        </>
    )
}

SinglesMenuGroup.propTypes = {
    category: PropTypes.object,
    onAddItem: PropTypes.func,
}

export default connect(null, {
    onAddItem: item => dispatch => {
        dispatch(tSetDashboardItems(item))
    },
})(SinglesMenuGroup)
