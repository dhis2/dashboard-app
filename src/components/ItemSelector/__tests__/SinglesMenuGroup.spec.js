import React from 'react';
import { shallow } from 'enzyme';
import { SinglesMenuGroup } from '../SinglesMenuGroup';

describe('SinglesMenuGroup', () => {
    const wrapper = props => shallow(<SinglesMenuGroup {...props} />);

    it('matches snapshot', () => {
        const props = {
            acAddDashboardItem: jest.fn(),
            category: {
                header: 'ponies',
                items: [
                    {
                        type: 'colorful',
                        name: 'Rainbow Dash',
                    },
                    {
                        type: 'greytone',
                        name: 'B&W',
                    },
                ],
            },
        };
        expect(wrapper(props)).toMatchSnapshot();
    });
});
