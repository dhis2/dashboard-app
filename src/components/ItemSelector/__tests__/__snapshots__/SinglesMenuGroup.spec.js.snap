// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SinglesMenuGroup matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SinglesMenuGroup
    acAddDashboardItem={[MockFunction]}
    category={
      Object {
        "header": "ponies",
        "items": Array [
          Object {
            "name": "Rainbow Dash",
            "type": "colorful",
          },
          Object {
            "name": "B&W",
            "type": "greytone",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <HeaderMenuItem
          title="ponies"
        />,
        Array [
          <ContentMenuItem
            name="Rainbow Dash"
            onInsert={[Function]}
            type="colorful"
          />,
          <ContentMenuItem
            name="B&W"
            onInsert={[Function]}
            type="greytone"
          />,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "title": "ponies",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "colorful",
        "nodeType": "function",
        "props": Object {
          "name": "Rainbow Dash",
          "onInsert": [Function],
          "type": "colorful",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "greytone",
        "nodeType": "function",
        "props": Object {
          "name": "B&W",
          "onInsert": [Function],
          "type": "greytone",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <HeaderMenuItem
            title="ponies"
          />,
          Array [
            <ContentMenuItem
              name="Rainbow Dash"
              onInsert={[Function]}
              type="colorful"
            />,
            <ContentMenuItem
              name="B&W"
              onInsert={[Function]}
              type="greytone"
            />,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": "ponies",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "colorful",
          "nodeType": "function",
          "props": Object {
            "name": "Rainbow Dash",
            "onInsert": [Function],
            "type": "colorful",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "greytone",
          "nodeType": "function",
          "props": Object {
            "name": "B&W",
            "onInsert": [Function],
            "type": "greytone",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
